app/controllers/items_controller.rb

```diff
@@ -0,0 +1,51 @@
+require 'pry'
+class ItemsController < ApplicationController
+  def index
+    @items = Item.all
+  end
+
+  def list
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def show
+    @item = Item.find(params[:id])
+  end
+
+  def new
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def create
+    @merchant = Merchant.find(params[:id])
+    Item.create(item_params)
+    redirect_to "/merchants/#{@merchant.id}/items"
+  end
+
+  def edit
+    @item = Item.find(params[:id])
+  end
+
+  def update
+    @item = Item.find(params[:id])
+    @item.update({
+      name: params[:name],
+      description: params[:description],
+      price: params[:price],
+      image: params[:image],
+      inventory: params[:inventory]
+      })
+    @item.save
+    redirect_to "/items/#{@item.id}"
+  end
```

Comment:
You have a method for your private params down below. You can use that method when you're both creating AND updating an entry in your database table. calling `.update` will also save this for you so you don't need to call .save. In addition to this, anytime you are redirecting from within a method, that means that you don't need to render a particular view for that method, so for that reason, you don't need instance variables. So, this method could be cleaned up to look like: 
```ruby 
  def update
    item = Item.find(params[:id])
    item.update(item_params)
    redirect_to "/items/#{item.id}"
  end
```